src/web_server/http_session.cc

```diff
@@ -163,28 +163,34 @@ struct http_session {
 
     // See if it is a WebSocket Upgrade
     if (boost::beast::websocket::is_upgrade(parser_->get())) {
-      // Disable the timeout.
-      // The websocket::stream uses its own timeout settings.
-      boost::beast::get_lowest_layer(derived().stream()).expires_never();
-
-      // Create a websocket session, transferring ownership
-      // of both the socket and the HTTP request.
-      return make_websocket_session(derived().release_stream(),
-                                    parser_->release(), settings_);
-    }
-
-    auto& queue_entry = queue_.add_entry();
-    if (settings_->http_req_cb_) {
-      settings_->http_req_cb_(
-          parser_->release(),
-          [self = derived().shared_from_this(),
-           &queue_entry](web_server::http_res_t&& res) {
-            std::visit(queue_entry, std::move(res));
-          },
-          derived().is_ssl());
+      if (!settings_->ws_upgrade_ok_ ||
+          settings_->ws_upgrade_ok_(parser_->get())) {
+        // Disable the timeout.
+        // The websocket::stream uses its own timeout settings.
+        boost::beast::get_lowest_layer(derived().stream()).expires_never();
+
+        // Create a websocket session, transferring ownership
+        // of both the socket and the HTTP request.
+        make_websocket_session(derived().release_stream(), parser_->release(),
+                               settings_);
```

Comment:
Is it safe to remove the `return` here? There is a `do_read()` call after the outer if/else block, which accesses the released/moved stream.

Reply:
Good catch!